core/vm/interpreter.go

```diff
@@ -58,7 +58,8 @@ func NewEVMInterpreter(evm *EVM) *EVMInterpreter {
 	switch {
 	case evm.chainRules.IsPrague:
 		// TODO replace with prooper instruction set when fork is specified
-		table = &shanghaiInstructionSet
+		// table = &shanghaiInstructionSet
+		table = &byzantiumInstructionSet
```

Comment:
This took me a while to figure out. I literally had to find which tx in the borked block had a gas usage different from Etherscan, and narrow the search down to the opcode generating the problem.

Was actually SLOAD, but this wasn't related to Prague fork but to the "previous" fork that we should use. The changed code makes sense for the shadow fork, but not for the replay data.